Poker.BE/Poker.BE.Domain/Game/Round.cs

```diff
@@ -25,69 +26,256 @@ public enum Move
         {
             switch (playMove)
             {
-                case Move.check :
+                case Move.check:
                     {
-                        currentTurn.Call();
+                        CurrentTurn.Check();  // Do nothing??
                         break;
                     }
-                case  Move.call:
+                case Move.call:
                     {
-                        currentTurn.Check();
+                        Call(amountToBetOrCall);
                         break;
                     }
                 case Move.fold:
                     {
-                        currentTurn.Fold();
-                        break;
+                        CurrentTurn.Fold();
+                        Player playerToRemove = this.CurrentPlayer;
+
+                        //Remover player from all pots
+                        Pot partialPotIterator = CurrentPot;
+                        while (partialPotIterator != null)
+                        {
+                            partialPotIterator.PlayersClaimPot.Remove(playerToRemove);
+                            partialPotIterator = partialPotIterator.PartialPot;
+                        }
+                        
+                        //Remove player from round
+                        this.currentPlayer = this.ActiveUnfoldedPlayers.ElementAt((ActiveUnfoldedPlayers.ToList().IndexOf(this.CurrentPlayer) - 1) % ActiveUnfoldedPlayers.Count);
+                        ActiveUnfoldedPlayers.Remove(playerToRemove);
+                        return;
+                        //break;
                     }
                 case Move.bet:
                     {
-                        currentTurn.Bet();
+                        //Check preconditions
+                        if (LastRaise > 0 || TotalRaise > 0)
+                            throw new IOException("Can't bet if someone had bet before... use raise move");
+                        if (TotalRaise + amountToBetOrCall == CurrentPlayer.Wallet.amountOfMoney + LiveBets[CurrentPlayer])
+                            throw new ArgumentOutOfRangeException("Can't bet all of your money... use all-in move");
+                        RaisePreconditions(amountToBetOrCall);
+
+                        //Make bet
+                        Raise(amountToBetOrCall);
+                        
                         break;
                     }
                 case Move.raise:
                     {
-                        currentTurn.Raise();
+                        //Check preconditions
+                        if (LastRaise == 0)
+                            throw new IOException("Can't raise if no one had bet before... use bet move");
+                        if (TotalRaise + amountToBetOrCall == CurrentPlayer.Wallet.amountOfMoney + LiveBets[CurrentPlayer])
+                            throw new ArgumentOutOfRangeException("Can't bet all of your money... use all-in move");
```

Comment:
`throw new ArgumentOutOfRangeException...` why not some `PokerException` extend class?!?!
![d4a](https://cloud.githubusercontent.com/assets/9889268/26259766/331dc24e-3cd3-11e7-81ee-31d78ae1af5a.gif)